src/m_menu.c

```diff
@@ -397,15 +398,19 @@ enum
 // associated with the item.
 //
 
+#define MAIN_MENU_RECT \
+    {0, 0, SCREENWIDTH, LINEHEIGHT}
+
 static menuitem_t MainMenu[]=
 {
-  {1,"M_NGAME", M_NewGame, 'n'},
-  {1,"M_LOADG", M_LoadGame,'l'},
-  {1,"M_SAVEG", M_SaveGame,'s'},
-  {1,"M_OPTION", M_Setup, 'o'}, // change M_Options to M_Setup
+  {1,"M_NGAME",  M_NewGame,  'n', NULL, 0, MAIN_MENU_RECT},
+  {1,"M_LOADG",  M_LoadGame, 'l', NULL, 0, MAIN_MENU_RECT},
+  {1,"M_SAVEG",  M_SaveGame, 's', NULL, 0, MAIN_MENU_RECT},
+  // change M_Options to M_Setup
+  {1,"M_OPTION", M_Setup,    'o', NULL, 0, MAIN_MENU_RECT}, 
   // Another hickup with Special edition.
-  {1,"M_RDTHIS",M_ReadThis,'r'},
-  {1,"M_QUITG", M_QuitDOOM,'q'}
+  {1,"M_RDTHIS", M_ReadThis, 'r', NULL, 0, MAIN_MENU_RECT},
+  {1,"M_QUITG",  M_QuitDOOM, 'q', NULL, 0, MAIN_MENU_RECT}
```

Comment:
Because of the `MainMenu[]` manipulations (reordering, enable/disable "Read This") a fully static definition of mrect_t won't work, so it somewhat defeats the purpose of these changes. Alternatively, we can set mrect_t in `M_Drawer` as before.

Is this PR approach okay?

Reply:
Honestly, I'm not even sure what this is supposed to fix.

Reply:
> Honestly, I'm not even sure what this is supposed to fix.

Try [hr2final.wad](https://www.doomworld.com/idgames/themes/hr/hr2final) with current master, it has interesting main menu 😆